File: backend/src/main/kotlin/com/example/backend/models/Depense.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import java.time.LocalDate
import java.time.LocalDateTime

@Entity
@Table(name = "depense")
class Depense {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY) // Use IDENTITY for auto-increment
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @Column(name = "caisse_id", length = 10)
  var caisseId: String? = null

  @Column(name = "designation")
  var designation: String? = null

  @Column(name = "quantite")
  var quantite: Int? = 0

  @Column(name = "prixUnitaire")
  var prixUnitaire: Int? = 0

  @Column(name = "dateDepense")
  var dateDepense: LocalDateTime? = null

  @Column(name = "beneficiaire", length = 100)
  var beneficiaire: String? = null

  @Column(name = "numeroCni", length = 100)
  var numeroCni: String? = null

  @Column(name = "dateDelivrance")
  var dateDelivrance: LocalDateTime? = null

  @Column(name = "lieuDelivrance", length = 100)
  var lieuDelivrance: String? = null

  @Column(name = "societe", length = 100)
  var societe: String? = null

  @Column(name = "typeDepense")
  var typeDepense: Int? = null

  @ColumnDefault("0")
  @Column(name = "supprimer")
  var  supprimer: Int? = 0
}

File: backend_py/pharmaPython/app/models/depense.py
from sqlalchemy import Column, Integer, String, DateTime
from app.core.db import Base

class Depense(Base):
  __tablename__ = "depense"

  id = Column(Integer, primary_key=True, index=True)
  caisse_id = Column(String(10))
  designation = Column(String)
  quantite = Column(Integer, default=0)
  prix_unitaire = Column(Integer, default=0)
  date_depense = Column(DateTime)
  beneficiaire = Column(String(100))
  numero_cni = Column(String(100))
  date_delivrance = Column(DateTime)
  lieu_delivrance = Column(String(100))
  societe = Column(String(100))
  type_depense = Column(Integer)
  supprimer = Column(Integer, default=0)
